tests/test_benchmark_constraints.py

```diff
@@ -261,6 +263,85 @@ def test_benchmark_minimal_degree_of_autonomy_constraint(self, margs):
+    # alone is called
+    @pytest.mark.skipif(
+        EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
+        reason="Benchmark test deactivated, set env variable "
+        "EXECUTE_TESTS_ON to 'master' to run this test",
+    )
+    @mock.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace())
+    def test_net_zero_energy_constraint(self, margs):
+        r"""
+        Notes
+        -----
+        With this benchmark test, the net zero energy (NZE) constraint is validated in a
+        single sector LES and a sector coupled LES.
+        Constraint_net_zero_energy_False contains a single sector LES without NZE constraint.
+        Constraint_net_zero_energy contains a single sector LES with NZE constraint.
+        Constraint_net_zero_energy_sector_coupled_False contains a sector-coupled LES without NZE constraint.
+        Constraint_net_zero_energy_sector_coupled contains a sector-coupled LES with NZE constraint.
+        The benchmark test passes if the degree of NZE of the defined energy systems
+        without constraint is lower than one and if the degree of NZE of the energy
+        systems with constraint equals one or is greater than one.
+
+        """
+        # define the cases needed for comparison
+        use_case = [
+            "Constraint_net_zero_energy_False",
+            "Constraint_net_zero_energy",
+            "Constraint_net_zero_energy_sector_coupled_False",
+            "Constraint_net_zero_energy_sector_coupled",
+        ]
+        # define empty dictionaries degree of NZE
+        degree_of_nze = {}
+        consumption_from_grid = {}
+        feedin_to_grid = {}
+        for case in use_case:
+            main(
+                overwrite=True,
+                display_output="warning",
+                path_input_folder=os.path.join(TEST_INPUT_PATH, case),
+                input_type=CSV_EXT,
+                path_output_folder=os.path.join(TEST_OUTPUT_PATH, case),
+            )
+            data = load_json(
+                os.path.join(
+                    TEST_OUTPUT_PATH, case, JSON_WITH_RESULTS + JSON_FILE_EXTENSION
+                )
+            )
+            degree_of_nze.update({case: data[KPI][KPI_SCALARS_DICT][DEGREE_OF_NZE]})
+            consumption_from_grid.update(
+                {
+                    case: data[KPI][KPI_SCALAR_MATRIX].set_index(LABEL)[
+                        ANNUAL_TOTAL_FLOW
+                    ]["Grid_DSO_consumption_source"]
+                }
+            )
+            feedin_to_grid.update(
+                {
+                    case: data[KPI][KPI_SCALAR_MATRIX].set_index(LABEL)[
+                        ANNUAL_TOTAL_FLOW
+                    ]["Grid_DSO_feedin_sink_sink"]
+                }
+            )
+
+        assert (
+            degree_of_nze[use_case[0]] < 1
+        ), f"The degree of NZE of a LES without NZE constraint should be lower than 1 but is {degree_of_nze[use_case[0]]}."
```

Comment:
You could state that is should be lower 1 so that the benchmark test reasonably confirms the feature (I mean, if you change some cost parameters you could anyway get to >= 1 here, and that would be correct but not useful)

Reply:
done with fcc8e7f